cmd/log-validator/main.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/hpcloud/tail"
+
+	"github.com/letsencrypt/boulder/cmd"
+	blog "github.com/letsencrypt/boulder/log"
+)
+
+func lineValid(text string) bool {
+	// Line format should match the following template:
+	//   timestamp hostname datacenter level binary-name[pid]: checksum msg
+	//
+	// template( name="LELogFormat" type="list" ) {
+	// 	property(name="timereported" dateFormat="rfc3339")
+	// 	constant(value=" ")
+	// 	property(name="hostname" field.delimiter="46" field.number="1")
+	// 	constant(value=" slc ")
+	// 	property(name="syslogseverity")
+	// 	constant(value=" ")
+	// 	property(name="syslogtag")
+	// 	property(name="msg" spifno1stsp="on" )
+	// 	property(name="msg" droplastlf="on" )
+	// 	constant(value="\n")
+	// }
+
+	fields := strings.Split(text, " ")
+	// Extract checksum from line
+	checksum := fields[5]
+	// Reconstruct just the message portion of the line
+	line := strings.Join(fields[6:], " ")
+	// Check the extracted checksum against the computed checksum
+	return checksum == blog.LogLineChecksum(line)
+}
+
+func validateFile(filename string) error {
```

Comment:
I think this function should probably return an error when there is at least one line in the file that isn't valid (in addition to printing the lines that aren't valid).

That way `main` can return non-zero when the validation of an explicit `-check-file` detected corrupted lines.

If you think it shouldn't for some reason can you add a comment to the function that makes it clearer that the only time `validateFile` will return an error is when the file can't be read?